components/learn/modules/ROOT/pages/write-contracts.adoc

```diff
@@ -0,0 +1,86 @@
+ * A set of https://consensys.github.io/smart-contract-best-practices/[best practices] was compiled by Consensys Dilligence: these are quite extensive, and include both https://consensys.github.io/smart-contract-best-practices/recommendations/[proven patterns] to learn from and https://consensys.github.io/smart-contract-best-practices/known_attacks/[known pitfalls] to avoid
+ * A rite of passage in the learning of Solidity is playing our very own https://solidity-05.ethernaut.openzeppelin.com/[Ethernaut] game: it'll test your understanding of both language and Ethereum concepts by having you look for subtle vulnerabilities in smart contracts as you advance through the increasingly difficult levels
+
+== Setting up a Solidity Project
+
+Simple works best: all you need to do is create directory called `contracts` and store your Solidity source files (`.sol`) there. That's it!
+
+All tools will automatically pick every contract in this directory, letting you organize your code in subdirectories (e.g. `admin`, `token`, etc.). To import a file from a different directory, simply refer to it by its relative path:
+
+```solidity
+import './admin/TransferrableAdmin.sol';
+```
+
+== Using the OpenZeppelin Contracts Library
+
+Reusable modules and libraries are the cornerstone behind great software. The https://openzeppelin.com/contracts/[*OpenZeppelin Contracts*] library is the result of years of community-driven effort towards a set of useful building blocks for smart contracts to build on. They've been the subject of multiple security audits, and their security and correctness battle-tested. And using them is as easy as any other contract!
+
+You'll first need to download the contracts from the latest published release:
+
+```bash
+$ npm install --save-dev @openzeppelin/contracts
+```
+
+Once installed, the library is ready to be used! Simply `import` the contracts as you would any other source in your system, prefixing the path with `@openzeppelin/contracts`:
+
+```solidity
+pragma solidity ^0.5.0;
+
+import '@openzeppelin/contracts/ERC20/ERC20.sol';
+
+contract MyToken is ERC20 {
+  constructor(uint256 initialSupply) public {
+    _mint(msg.sender, initialSupply);
+  }
+}
+```
+
+The OpenZeppelin Contracts feature extensive https://docs.openzeppelin.com/contracts/2.x/#next-steps[documentation], complete with a detailed API reference: see for example the entry for the https://docs.openzeppelin.com/contracts/2.x/api/token/erc20#ERC20-_mint-address-uint256-[`_mint` function] used in the code sample above.
+
+While the library is designed in such a way that it is difficult to misuse it, when in doubt, you can always refer to the documentation for clarification.
+
+== Compiling your Contracts
+
+Your contracts are set up, libraries installed, but how do you actually transform these Solidity source files into executable Ethereum bytecode?
```

Comment:
It's not obvious that "transform into executable bytecode" is what "compile" means. I would add a sentence saying something like "before we deploy we have to compile the source code into EVM bytecode".